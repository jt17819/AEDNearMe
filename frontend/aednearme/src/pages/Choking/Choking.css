#choking-link{
    color: rgb(142, 255, 67);
}