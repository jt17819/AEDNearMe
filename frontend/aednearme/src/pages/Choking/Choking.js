import { Typography, Button, Grid, Container, makeStyles } from '@material-ui/core'
import React , { Fragment } from 'react'
import { useNavigate } from 'react-router';
import ChokingVid from '../../components/ChokingVid/ChokingVid';
import './Choking.css'

const useStyles = makeStyles(() => ({

  text: {
    fontSize: '16px'
  },

  btn_box:{
    display: 'flex',
    justifyContent: 'space-between',
  },

  primary: {
    border: "none",
    background: "#273746",
    color: '#fff',
    borderRadius: "0px",
    padding: "10px 20px",
    "&:hover": {
      background: "#34495E",
    }
  },
  
  secondary: {
    border: "none",
    background: "aliceblue",
    color: "#ba181b",
    borderRadius: "0px",
    padding: "10px 20px",
    "&:hover": {
      background: "aliceblue",
    }
  },

  

}));


const Choking = () => {

  const navigate = useNavigate();
  const goToCPRhowTo = () => {
    navigate('/CPRhowTo')
  }

  const classes = useStyles();

  
  return (

    <Fragment>

    <Container style={{
      paddingBottom: "80px",
      paddingTop: "5px"
      }}> 

      <h1 className={classes.title} >Learn first aid for someone who is choking</h1>

        <Grid container spacing={1}>
          <Grid item md={5} >
            <h3 className={classes.text}>
            Step 1: If someone is choking, encourage them to cough.
            </h3>
              <p>If the blockage is severe, they may be holding their chest or neck and won't be able to speak, breathe or cough, and you will need to help them.
              </p>

            <h3 className={classes.text}>
            Step 2: Bend them forwards and give up to 5 back blows to try and dislodge the blockage.
            </h3>
              <p>Hit them firmly on their back with the heel of your hand between the shoulder blades.
              </p>

            <h3 className={classes.text}>
            Step 3: If they are still choking, give up to 5 abdominal thrusts: hold around the waist and pull inwards and upwards above their navel.
            </h3>

            <h3 className={classes.text}>
            Step 4: If they are still choking and become unresponsive, follow the procedure on how to give CPR.
            </h3>


              <Button variant="contained" 
                className={classes.primary}  
                onClick={goToCPRhowTo}
              >CPR: How to give</Button>


            <h3>Read more here at <a id="choking-link" href="https://www.sja.org.uk/get-advice/first-aid-advice/choking/" target="_blank">St. John's Ambulance</a>
            </h3>

          </Grid> 

          <Grid
            item
            md={7}
            xs={12}
            className=""
            >
            <div>
              <ChokingVid />
            </div>  

          </Grid>

        </Grid>
      </Container>
  </Fragment>

       
        
  )
}

export default Choking;