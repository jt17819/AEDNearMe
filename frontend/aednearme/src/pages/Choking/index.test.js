import { render, fireEvent, screen, cleanup, within } from '@testing-library/react';
import { MemoryRouter, Router as Router } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'
import {default as Choking} from './Choking';
import axios from 'axios'

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe('Choking', () => {

  // beforeEach(() => {
  //   render(<App />, { wrapper: MemoryRouter })
  // });


  // Rendering

  it('renders welcome message', () => {
    render(<Choking />, { wrapper: MemoryRouter })
    expect(screen.getByText('Learn first aid for someone who is choking')).toBeInTheDocument();
  });

  test('list contains 1 buttons', () => {
    render(<Choking />, { wrapper: MemoryRouter })

  
    // const listElement = screen.getAllByRole('button');
    const listItems = screen.getAllByRole('button');
  
    
    expect(listItems.length).toEqual(1);
  });



  // Snapshot

  it('should take a snapshot', () => {
    render(<Choking />, { wrapper: MemoryRouter })
    const { asFragment } = render(<Choking />, { wrapper: MemoryRouter })
    expect(asFragment(<Choking />)).toMatchSnapshot()
  })

  

  
})

