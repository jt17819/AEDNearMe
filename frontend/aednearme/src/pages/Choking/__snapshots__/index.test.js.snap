// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choking should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root pb-5 MuiContainer-maxWidthLg"
    style="padding-bottom: 70px; padding-top: 10px;"
  >
    <h1
      class="makeStyles-title-11"
    >
      Learn first aid for someone who is choking
    </h1>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-5"
      >
        <h3
          class="makeStyles-text-12"
        >
          Step 1: If someone is choking, encourage them to cough.
        </h3>
        <p>
          If the blockage is severe, they may be holding their chest or neck and won't be able to speak, breathe or cough, and you will need to help them.
        </p>
        <h3
          class="makeStyles-text-12"
        >
          Step 2: Bend them forwards and give up to 5 back blows to try and dislodge the blockage.
        </h3>
        <p>
          Hit them firmly on their back with the heel of your hand between the shoulder blades.
        </p>
        <h3
          class="makeStyles-text-12"
        >
          Step 3: If they are still choking, give up to 5 abdominal thrusts: hold around the waist and pull inwards and upwards above their navel.
        </h3>
        <h3
          class="makeStyles-text-12"
        >
          Step 4: If they are still choking and become unresponsive, follow the procedure on how to give CPR.
        </h3>
        <div
          class="makeStyles-btn_box-13"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-primary-14"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              CPR: How to give
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <h3>
          Read more here at 
          <a
            href="https://www.sja.org.uk/get-advice/first-aid-advice/choking/"
            id="choking-link"
            target="_blank"
          >
            St. John's Ambulance
          </a>
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <div>
          <div>
            <div
              style="width: 100%; height: 360px;"
            >
              <div
                style="width: 100%; height: 100%;"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
